core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -445,6 +376,59 @@ object Endpoint {
     final def apply(mapper: Mapper[F, A]): Endpoint[F, mapper.Out] = mapper(self)
   }
 
+  /** */
+  trait Validatable[F[_], A] extends Endpoint[F, A] { self =>
```

Comment:
Agreed. We have a Mappable thing that behaves similarly but yea, it has its limitations. 

I think, the most reasonable thing to do might be actually to accept a validation rule as part of endpoint construction. Something along the lines:

```
// before
param[Int]("foo").should(beLessThan(10)) // to much sugar

// after
param[Int]("foo", Validation.lessThan(10))
```

WDYT?

Reply:
Also related: #808

Reply:
@vkostyukov I think I like your suggestion more. Bonus points if we can create a `Validation` from a function that returns `cats.Validated` 😄

Reply:
I changed my mind and ripped validations off the API altogether. I'm starting to think we shouldn't be in the business of validating things and it should purely be an application concern, where users can fully leverage `cats.Validated` (and any other validation machinery) directly. Endpoints are fully-typed so nothing stopped them to have `Endpoint[IO, Validated[Foo, Bar]]` types in their domain.

I also like the idea of doing less in the APIs. 

I know it's a breaking API change but I'm personally willing to break things once to get a reasonable 1.0 out of the doors.

Reply:
>  it should purely be an application concern

May I suggest removing `ToAsync` as well then? 😇 I don't think HTTP library should deal with effect transformations.
It'll simplify mapper a bit

Reply:
I'd be down! I think we still need it when we bootstrap HTTP server and want to "await" on the server closure but maybe it wouldn't be too bad for folks to write it themself?

Reply:
As far as they can provide an instance of `Async` for their `F[_]`, we shouldn't care?

Reply:
>  we bootstrap HTTP server and want to "await" on the server closure

Tbh, lifecycle management is another topic. I'd see finch-core returning a resource with a service given `F[_] : Async` boundary, and then let users decide how they want to bind it to port.

Could be another subproject that provides additional capabilities such as `FinchApp`, although more often then not I find myself hacking all these `*App` classes.

Reply:
Yeah I'd rather be in favour of removing `ToAsync` (isn't it basically the same as `Dispatcher`?)
I'm less sure about validation because I have no estimation of how widely it's used.

Reply:
Removing ToAsync: https://github.com/finagle/finch/pull/1480

Reply:
For what it's worth, we've never used validation API and always used mapping in our company.
So I would support the breaking change of removal of validation API based on my biased opinion 😄

Reply:
I think that's what happens in practice - people roll their own validation. The existing API wasn't that good anyway. Specifically, when you compose validation rules, you lose the integrity: if `a or b` fails, we won't tell you which one is it.